pkgs/by-name/fi/filebot/package.nix

```diff
@@ -63,9 +70,25 @@ stdenv.mkDerivation (finalAttrs: {
       --replace '$FILEBOT_HOME/data/.license' '$APP_DATA/.license' \
       --replace '-jar "$FILEBOT_HOME/jar/filebot.jar"' '-Dcom.googlecode.lanterna.terminal.UnixTerminal.sttyCommand=${coreutils}/bin/stty -jar "$FILEBOT_HOME/jar/filebot.jar"'
     wrapProgram $out/opt/filebot.sh \
-      --prefix PATH : ${lib.makeBinPath [ openjdk17 ]}
+      --prefix PATH : ${
+        lib.makeBinPath [
+          chromaprint
+          libmediainfo
+          openjdk17
+          openjfx
+          p7zip
+          unrar
+          zenity
+        ]
+      }
     # Expose the binary in bin to make runnable.
     ln -s $out/opt/filebot.sh $out/bin/filebot
+
+    # Delete the built-in libmediainfo so the program has to use the nixpkgs version
+    rm $out/opt/lib/**/libmediainfo.so
+    ln -s ${libmediainfo.outPath}/lib/libmediainfo.so $out/opt/lib/Linux-aarch64/libmediainfo.so
+    ln -s ${libmediainfo.outPath}/lib/libmediainfo.so $out/opt/lib/Linux-armv7l/libmediainfo.so
+    ln -s ${libmediainfo.outPath}/lib/libmediainfo.so $out/opt/lib/Linux-x86_64/libmediainfo.so
```

Comment:
It's a bit hacky but not *too* bad.

Have you tried instead removing that file instead and adding `lib.getLib libmediainfo` to the rpath via patchelf instead? That'd strike me as cleaner.

Reply:
I did not try that. That's a new one to me. I'll look it up and try tonight. I did try just removing it and hoping the program would find the nixpkgs version but that did not work. Probably because of the rpath thing you mentioned.

Reply:
Yeah there is no global Nix library path, so binaries' rpaths need to be adjusted to include the lib's path.

Reply:
I did try to learn about patchelf and the rpath, but then it occurred to me: this is a java program, and patchelf patches, well, elf files. It shouldn't work on this, right? I see there's an autopatchelfhook in the file already that contains the libmediainfo, but it doesn't actually make it work without the symlinks. I tried removing them and just leaving the 'rm' to remove the broken lin, but it fails to load a lib at all, stating that it's missing.

Reply:
Right that's a good point.

How is it loading the dynamic libs then? JNI? I'm not really familiar with how that works or how it could be made to have a library search path. Like, what decides that it looks in that opt dir? It's a store path, so I guess it's dynamic based on executable location (argv[0]?)?